.changeset/twelve-coats-rush.md

```diff
@@ -0,0 +1,35 @@
+---
+'astro': major
+---
+
+The endpoints that have **lowercase** names are **deprecated**.
+
+The endpoints methods **need** to be in **uppercase**.
+
+```diff
+--export function get() {
+++export function GET() {
+    return new Response(JSON.stringify({ "title": "Bob's blog" }));
+}
+
+--export function post() {
+++export function POST() {
+    return new Response(JSON.stringify({ "title": "Bob's blog" }));
+}
+
+--export function put() {
+++export function PUT() {
+    return new Response(JSON.stringify({ "title": "Bob's blog" }));
+}
+
+--export function all() {
+++export function ALL() {
+    return new Response(JSON.stringify({ "title": "Bob's blog" }));
+}
+
+// you can use use the whole word "DELETE"
+--export function del() {
+++export function DELETE() {
+    return new Response(JSON.stringify({ "title": "Bob's blog" }));
+}
+```
```

Comment:
I’m not sure we usually use this double `--`/`++` syntax in diffs? Probably safest to stick with a single `-`/`+` at the start of the line. Also fixed the repeated “use” in the comment above `DELETE`.

````suggestion
```diff
- export function get() {
+ export function GET() {
    return new Response(JSON.stringify({ "title": "Bob's blog" }));
}

- export function post() {
+ export function POST() {
    return new Response(JSON.stringify({ "title": "Bob's blog" }));
}

- export function put() {
+ export function PUT() {
    return new Response(JSON.stringify({ "title": "Bob's blog" }));
}

- export function all() {
+ export function ALL() {
    return new Response(JSON.stringify({ "title": "Bob's blog" }));
}

// you can use the whole word "DELETE"
- export function del() {
+ export function DELETE() {
    return new Response(JSON.stringify({ "title": "Bob's blog" }));
}
```
````